packages/server/src/modules/activity/middleware.ts

```diff
@@ -63,24 +63,20 @@ export const findActivities: MiddlewareFunction = (req, res, next) => {
 export const canEditActivities =
   (key: string): MiddlewareFunction =>
   (req, res, next) => {
-    const canEdit = service.canEditActivities(
-      res.locals[key],
-      res.locals.me.id,
-    );
+    const canEdit = service.canEditActivities(res.locals[key], res.locals.me);
     if (!canEdit) {
       throw new UnauthorizedActivityEditError();
     }
     next();
   };
 
-export const sendActivity: MiddlewareFunction = (req, res, next) => {
+export const sendActivity: MiddlewareFunction = async (req, res, next) => {
   const activity = res.locals.activity as Activity;
   if (!activity) {
     throw new ActivityNotPreparedError();
   }
   // TODO - verify if this is working
-  [...activity.roles]
-    .map(({ user }) => user)
+  (await application.service.getUsersWithRole(activity, 'Participator'))
```

Comment:
Because of the permission inheritance all users with higher level of permission will also get the message?

Reply:
When querying for user permission on action, stronger role will inherit it as well, but when querying for users with specific role, you only get explicitly assigned. 

It's not set in stone. I'm still in the process of figuring out what's needed from the permission API.

Reply:
Ok, in this case then I think all the users associated with this subject should get activity. 

Maybe it makes sense to have a different signature which allows omitting role to get all associated users. Another option is to create a new service for that purpose.

TLDR; we want sometimes (in this case) to notify all associated users.